From: amodra@bigpond.net.au
Subject: http://sources.redhat.com/bugzilla/show_bug.cgi?id=1042

/usr/bin/ld: arch/ppc64/kernel/vdso32/vdso32.so: The first section in the PT_DYNAMIC segment is not the .dynamic section

Signed-off-by: Olaf Hering <olh@suse.de>

 arch/ppc64/kernel/vdso32/vdso32.lds.S |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6.12/arch/ppc64/kernel/vdso32/vdso32.lds.S
===================================================================
--- linux-2.6.12.orig/arch/ppc64/kernel/vdso32/vdso32.lds.S
+++ linux-2.6.12/arch/ppc64/kernel/vdso32/vdso32.lds.S
@@ -40,9 +40,9 @@ SECTIONS
   .gcc_except_table	: { *(.gcc_except_table) }
   .fixup		: { *(.fixup) }
 
-  .got ALIGN(4)		: { *(.got.plt) *(.got) }
-
   .dynamic		: { *(.dynamic) }		:text	:dynamic
+  .got : { *(.got) }
+  .plt : { *(.plt) }
 
   _end = .;
   __end = .;
